apps/api/src/common/graphql/services/core.service.ts

```diff
@@ -57,12 +59,27 @@ export abstract class CoreService<TEntity> {
     options.filters?.forEach((filter, index) => {
       const field = filter.field;
       const operator = filter.operator;
-      const value =
-        this.isDateField(field) && (operator === 'gt' || operator === 'lt')
-          ? new Date(filter.value)
-          : filter.value;
       const paramName = `param${index}`;
       let condition = `${alias}.${field}`;
+      let value: FilterValue = filter.value;
+
+      if (this.isDateField(field) && (operator === 'gt' || operator === 'lt')) {
+        value = new Date(filter.value as string);
+      } else if (operator === 'in') {
+        if (typeof filter.value === 'string') {
+          try {
+            value = JSON.parse(filter.value);
+
+            if (!Array.isArray(value)) {
+              value = filter.value.split(',').map((item) => item.trim());
+            }
+          } catch (error) {
```

Comment:
For your query, If the values are already a string, separated by commas, can we not just pass it directly pass it in the query statement?
I applied the approach because then the input values are consistently formatted, validated, and compatible with the expected query.
 
The try-catch block is used to handle cases where filter.value is a string but not a valid JSON string. So, even if filter.value cannot be parsed as JSON, the code can still handle it by treating it as a comma-separated list.

Reply:
But what will happen if there is some other error? What you have mentioned is a very specific-case error that the string is valid comma separated list.

Reply:
Outdated code and suggestion, resolving conversation